src/compressed_chunk.c

```diff
@@ -32,6 +34,89 @@ void Compressed_FreeChunk(Chunk_t *chunk) {
   free(chunk);
 }
 
+static void swapChunks(CompressedChunk *a, CompressedChunk *b) {
+  CompressedChunk tmp = *a;
+  *a = *b;
+  *b = tmp;
+}
+
+ChunkResult Compressed_UpsertSample(AddCtx *aCtx) {
+  ChunkResult res;
+  ChunkResult rv = CR_OK;
+  CompressedChunk *oldChunk = (CompressedChunk *)aCtx->inChunk;
+
+  short newSize = oldChunk->size / sizeof(Sample);
```

Comment:
Size of the new chunk to which we insert all samples

Reply:
@ashtul you are calculating the ratio between the chunk size and the sample size (fixed 16 bytes). 
so the new size will be n/16 of the previous chunk size?

Reply:
No.
The size we pass is by samples. We should consider changing that outside of the scope of Oot-of-order as it is confusing.
Historically, `size` was more of `numSamples`

Reply:
@danni-m Should make another PR to straighten this?
We should also think about changing `CHUNK_SIZE` argument which is confusing especially since we expect most users to use compressed chunks